src/config/discover.ml

```diff
@@ -0,0 +1,48 @@
+module C = Configurator.V1
+
+let replace_last search replace_with input =
+  let input_len = String.length input in
+  let search_len = String.length search in
+  let rec loop start last_pos =
+    if start >= input_len then
+      match last_pos with
+      | None -> input
+      | Some pos ->
+        String.sub input 0 pos ^ replace_with ^ String.sub input (pos + search_len) (input_len - pos - search_len)
+    else
+      try
+        let pos = String.index_from input start search.[0] in
+        if pos >= 0 && pos + search_len <= input_len &&
+           String.sub input pos search_len = search
+        then
+          loop (pos + search_len) (Some pos)
+        else
+          loop (pos + 1) last_pos
+      with Not_found ->
+        match last_pos with
+        | None -> input
+        | Some pos ->
+          String.sub input 0 pos ^ replace_with ^ String.sub input (pos + search_len) (input_len - pos - search_len)
+  in
+  loop 0 None
+
+let () =
+  C.main ~name:"lzo-pkg-config" (fun c ->
+      let default : C.Pkg_config.package_conf =
+        { libs = ["-llzo2"]; cflags = [] }
+      in
+      let conf =
+        match C.Pkg_config.get c with
+          | None -> default
+          | Some pc -> (
+              match
+                C.Pkg_config.query pc ~package:"lzo2"
+              with
+                | None -> default
+                | Some deps -> deps)
+      in
+      (* Fixes the Homebrew quirk that returns include paths containing library name,
+          which prevents including using the "<lzo/lzo1x.h>" syntax *)
+      let cflags = List.map (replace_last "/include/lzo" "/include") conf.cflags in
+      C.Flags.write_sexp "c_flags.sexp" cflags;
```

Comment:
That seems to be a bug in the binding instead of pkg-config. The includes in the C code should be devoided of the `lzo/` prefix. On Linux I'm also getting the exact same thing as in macOS:
```
$ pkg-config --cflags lzo2
-I/usr/include/lzo 
```
```suggestion
let () =
  C.main ~name:"lzo-pkg-config" (fun c ->
      let default : C.Pkg_config.package_conf =
        { libs = ["-llzo2"]; cflags = [] }
      in
      let conf =
        match C.Pkg_config.get c with
          | None -> default
          | Some pc -> (
              match
                C.Pkg_config.query pc ~package:"lzo2"
              with
                | None -> default
                | Some deps -> deps)
      in
      C.Flags.write_sexp "c_flags.sexp" conf.cflags;
```

Reply:
@kit-ty-kate at first I tried to just fix the source code, the problem is that LZO headers themselves refer to other parts using the `#include <lzo/...>` directive, thus it causes error like this one:
```
ℤ dune build
File "src/dune", line 9, characters 9-18:
9 |   (names lzo_stubs)
             ^^^^^^^^^
In file included from lzo_stubs.c:2:
/opt/homebrew/Cellar/lzo/2.10/include/lzo/lzo1x.h:33:10: fatal error: 'lzo/lzoconf.h' file not found
#include <lzo/lzoconf.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

Reply:
The code that includes this defines it like this:
```
#ifndef __LZOCONF_H_INCLUDED
#include <lzo/lzoconf.h>
#endif
```
So you just need to add the missing `#include <lzoconf.h>` before the other include and it'll work.
All the examples in the source code of lzo2 has this pattern.

Reply:
Thanks for pointing this out! It helped.